proposals/NativeInt.md

```diff
@@ -0,0 +1,263 @@
+# Native-sized integers
+
+## Summary
+[summary]: #summary
+
+Language support for a native-sized signed and unsigned integer types.
+
+The motivation is for interop scenarios and for low-level libraries.
+
+## Design
+[design]: #design
+
+New contextual keywords `nint` and `nuint` represent native signed and unsigned integer types.
+The identifiers are only treated as keywords when used in a type context and when the identifier does not otherwise bind to a symbol at that program location.
+
+The types `nint` and `nuint` are essentially aliases to underlying types `System.IntPtr` and `System.UIntPtr`, where the compiler can surface additional conversions and operations for native ints.
```

Comment:
Presumably a secondary motivation is to reduce confusion? I have always found it confusing that `IntPtr` is often not representing a pointer.